src/backend/distributed/commands/text_search.c

```diff
@@ -99,6 +107,42 @@ PostprocessCreateTextSearchConfigurationStmt(Node *node, const char *queryString
 }
 
 
+/*
+ * PostprocessCreateTextSearchDictionaryStmt is called after the TEXT SEARCH DICTIONARY has been
+ * created locally.
+ */
+List *
+PostprocessCreateTextSearchDictionaryStmt(Node *node, const char *queryString)
+{
+	DefineStmt *stmt = castNode(DefineStmt, node);
+	Assert(stmt->kind == OBJECT_TSDICTIONARY);
+
+	if (!ShouldPropagate())
+	{
+		return NIL;
+	}
+
+	/* check creation against multi-statement transaction policy */
+	if (!ShouldPropagateCreateInCoordinatedTransction())
+	{
+		return NIL;
+	}
+
+	EnsureCoordinator();
+	EnsureSequentialMode(OBJECT_TSDICTIONARY);
+
+	ObjectAddress address = GetObjectAddressFromParseTree((Node *) stmt, false);
+	EnsureDependenciesExistOnAllNodes(&address);
+
+	List *commands = CreateTextSearchDictDDLCommandsIdempotent(&address);
```

Comment:
I think it is ok to take this approach, I am just wondering, is there a specific reason for text search dictionaries why we want to recreate the statements from the catalog instead of using the statement that is passed around in the utility hook?

For text search configuration's it was mostly due to the most common way of creating a configuration is via the COPY argument which copies all settings from an existing configuration. The one copying from might not have been a distributed one. Instead of distributing that one so that we could copy the configuration I decided to use this pattern there.

From my quick read on the postgresql documentation it doesn't seem to be the same for text search dictionaries. Are there any downsides of passing the COMMENT  and OWNERSHIP commands always like this?

No need to change, mostly curious.

I was recently contemplating if this would be the easier approach for many objects as well, which could help with reducing the 'duplicate' code feeling many of the object propagation implementations have recently. Understanding this use here better could help me inform on that topic.

Reply:
No particular reason. I thought that it is ok to create the SQL from catalog entries as well as using the current one after qualifying everything.

Now when I think about it, your suggestion made more sense to me to just qualify the current sql command as it is the simpler thing to do.

Reply:
Thanks!